resources/js/data-access/memo-api/resources/member.resource.ts

```diff
@@ -0,0 +1,9 @@
+export type MemberResource = {
+  userId: string;
+  facilityId: string;
+  hasFacilityAdmin: boolean;
+  /**
+   */
+  staffMemberId: string | null;
```

Comment:
Może by undefined używać i udawać że `null` nigdy nie istniał? Ja tak robię, nigdy nie kupowałem tego o intentionally empty, dwa nulle to po prostu kolejna z rzeczy które zrobili źle jak tworzyli ten język :) Ale bardzo się nie upieram.

Reply:
też nie lubię podziału na null i undefined, php nawet nie wie co to undefined, a js ma 2 odpowiedzi, czym jest undefined:
- pole nie istnieje w obiekcie
- pole ma wartość undefined
Dla spokoju bym zachował, że to, co jest ustawione na BE jako nullable, ma w tsie `type | null`;
Czasem może być tak, że backend może oznaczyć pole zwracane jako opcjonalne (np. doda parametr include, który po ustawieniu true dopiero zwróci to pole, a jak będzie false to nawet tego pola nie będzie w zwrotce - to jest przypadek z possibly undefined field).
Czyli:
- pole nullable: field: type | null;
- pole opcjonalne: field?: type;

Reply:
Nie będę się jakoś bardzo kłócić ale wydaje mi się że w tym układzie prędzej czy później się okaże że musimy rozróżnić null i undefined/brak, a to będzie znaczyło że nie zrobiliśmy tego optymalnie. Ale pewnie przeżyjemy.